relay-server/src/actors/healthcheck.rs

```diff
@@ -36,7 +33,7 @@ impl Healthcheck {
     ///
     /// Panics if the service was not started using [`Healthcheck::start`] prior to this being used.
     pub fn from_registry() -> Addr<Self> {
-        ADDRESS.read().as_ref().unwrap().clone()
+        REGISTRY.get().unwrap().healthcheck.clone()
```

Comment:
I wonder if we could make this method generic on `Service` somehow. In C++ you could do `std::get<T>` to access a tuple by type, in Rust I would not know how to do it.

Reply:
yeah we wondered this as well but could not figure it out without dynamic dispatch (`TypeId`) but that would slow this down.  I'm not sure how to do this at compile time, but it so feels like this should be possible.